src/pages/index.js

```diff
@@ -0,0 +1,393 @@
+    });
+}
+
+// todo - implement loading text for all other form submissions
+
+function handleAddCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const addCardModalSubmitBtn = evt.submitter;
+  addCardModalSubmitBtn.textContent = 'Saving...';
+  setButtonText(addCardModalSubmitBtn, true, "Save", "Saving...");
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      console.log(data);
+
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+    }).catch(console.error)
+    .finally(() => {
+
+      addCardModalSubmitBtn.textContent = "Save";
+    });
+  disableButton(cardSubmitBtn, settings);
+  closeModal(cardModal);
+
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSubmitBtn = evt.submitter;
+
+  avatarSubmitBtn.textContent = 'Saving...';
+  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+
+  console.log(avatarInput.value);
+  // const formData = new FormData(avatarForm);
+  // const avatarUrl = formData.get('profile-avatar-input');
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      console.log(data);
+      // const avatarImageSrc = document.getElementById("profile-avatar");
+      avatarImageSrc.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      avatarSubmitBtn.textContent = "Save";
+    });
+  evt.target.reset();
+  disableButton(avatarSubmitBtn, settings);
+}
+
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const deleteFormSubmitBtn = evt.submitter;
+  setButtonText(deleteFormSubmitBtn, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      setButtonText(deleteFormSubmitBtn, false, "Delete", "Deleting...");
+    });
+}
+
+
+function handleLike(btn, id) {
+  // evt.preventDefault();
+  //   cardLikeBtn.addEventListener('click', (evt) => {
+  //   handleLike(evt, data._id);
+  // });
+
+  const isLiked = btn.classList.contains('card__like-btn_liked');
+  changeLikeState(btn, id, isLiked);
+}
+function changeLikeState(btn, id, isLiked) {
+  api.handleLike(id, isLiked)
+    .then((data) => {
+      btn.classList.toggle('card__like-btn_liked', data.isLiked);
+    }
+    )
+    .catch(console.error);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector('.card')
+    .cloneNode(true);
+  const cardNameEl = cardElement.querySelector('.card__title');
+  const cardImageEl = cardElement.querySelector('.card__image');
+  const cardLikeBtn = cardElement.querySelector('.card__like-btn');
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
```

Comment:
* When I reload the page, all **red** hearts become **transparent** again.


Please, check `isLiked` here, and add the color